### ISSUE TITLE
Not issue but clarify question

### ISSUE THREAD

**Issue by cleargoal** [sanduhrs/phpstorm-url-handler#21]
I am forced to write here because this repo doesn't have a 'discussions' tab.

In the doc there is the sentence:

> Then copy the actual handler to your $PATH and make it executable.

Please explain what the `actual handler` is in this mention.

Thank you!

**Comment by rozwell** [comment]
> Then copy the actual handler to your $PATH and make it executable.

Instead of "Then copy" there should be "Or copy" because it's the 3rd alternative and it describes this operation:
```
cp phpstorm-url-handler /usr/bin/phpstorm-url-handler
chmod +x /usr/bin/phpstorm-url-handler
```
Which will copy actual (handler) file `phpstorm-url-handler` to the `/usr/bin/` directory and make it executable.

**Comment by cleargoal** [comment]
> Instead of "Then copy" there should be "Or copy" because it's the 3rd alternative and it describes this operation:
> 
> ```
> cp phpstorm-url-handler /usr/bin/phpstorm-url-handler
> chmod +x /usr/bin/phpstorm-url-handler
> ```
> Which will copy actual (handler) file `phpstorm-url-handler` to the `/usr/bin/` directory and make it executable.

Thanks for your answer, but it is still not clear to me. Where is or would be the file `phpstorm-url-handler`?

**Comment by rozwell** [comment]
> Where is or would be the file `phpstorm-url-handler`?

That's the main file in this repository: https://github.com/sanduhrs/phpstorm-url-handler/blob/master/phpstorm-url-handler
Notice: to work it requires `phpstorm` to be available in your system `$PATH` or as an alias.

It's a simple bash script and it's code should be self explanatory.

**Comment by cleargoal** [comment]
> > Where is or would be the file `phpstorm-url-handler`?
> 
> That's the main file in this repository: https://github.com/sanduhrs/phpstorm-url-handler/blob/master/phpstorm-url-handler Notice: to work it requires `phpstorm` to be available in your system `$PATH` or as an alias.
> 
> It's a simple bash script and it's code should be self explanatory.

**Thank you for the help!**
I'm very appreciate it!

This is my PATH
/home/yefrem/.nvm/versions/node/v20.12.2/bin:/home/yefrem/.local/share/pnpm:/home/yefrem/.local/bin:/home/yefrem/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/yefrem/.composer/vendor/bin:::::: /opt/PhpStorm-233.11799.297/bin:/opt/PhpStorm-233.11799.297/bin

Does it look correct?


I added phpStorm by this line in bashrc
`export PATH="$PATH:/opt/PhpStorm-233.11799.297/bin"`

One more question.
This package has no installation command.
So, do I need simply to put `phpstorm-url-handler` into the `/usr/bin/` directory?

**Comment by rozwell** [comment]
> This is my PATH /home/yefrem/.nvm/versions/node/v20.12.2/bin:/home/yefrem/.local/share/pnpm:/home/yefrem/.local/bin:/home/yefrem/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/yefrem/.composer/vendor/bin:::::: /opt/PhpStorm-233.11799.297/bin:/opt/PhpStorm-233.11799.297/bin
> 
> Does it look correct?
>
> I added phpStorm by this line in bashrc `export PATH="$PATH:/opt/PhpStorm-233.11799.297/bin"` 

Yes, apart from few empty entries and duplicated `/opt/PhpStorm-233.11799.297/bin` - which are probably caused by [putting `$PATH` in `.bashrc`](https://unix.stackexchange.com/questions/476593/when-should-i-use-bashrc-and-when-profile).

Move your **$PATH** to the end of `~/.profile` without `export`, like so:
```bash
PATH="/opt/PhpStorm-233.11799.297/bin:$PATH"
```
Notice: order in `$PATH` matters and whatever is first, will be executed first.

> One more question. This package has no installation command. So, do I need simply to put `phpstorm-url-handler` into the `/usr/bin/` 

That's correct, if you want it to be available for the entire system.

Actually, you can put it wherever you want and simply add the directory to `$PATH`.
I suggest storing custom executables in your home directory like `~/bin` so it doesn't get deleted during a system upgrade and is usually added automatically to `$PATH` if that directory exists.
Here is a part of `~/.profile` responsible for that:
```bash
# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/bin" ] ; then
    PATH="$HOME/bin:$PATH"
fi
```
If it doesn't work (don't forget to open new bash session or relog), just add it to your `$PATH`:
```bash
PATH="/opt/PhpStorm-233.11799.297/bin:$HOME/bin:$PATH"
```

**Comment by cleargoal** [comment]
> Move your $PATH to the end of ~/.profile without export, like so:

**Thank you very much!** 
It's done and looks normal at my comp now.
You are a good person!

**Comment by cleargoal** [comment]
> I suggest storing custom executables in your home directory like ~/bin

Sorry for the silly questions, but please look at the screenshot
![image](https://github.com/user-attachments/assets/51ab4d70-4402-4acd-955f-3fa313ae343d)
Is `/usr/bin` the directory you mean?

**Comment by rozwell** [comment]
> You are a good person!

Thank you ;)


> Is `/usr/bin` the directory you mean?

You missed the `~` (tilde) sign.
Path `~/` means user home directory, which in your case will be automagically translated to `/home/yefrem/`

Give it a try: 
```bash
realpath ~/
```

**Comment by rozwell** [comment]
@cleargoal It's time to close this issue, don't you think? ;)

**Comment by cleargoal** [comment]
> @cleargoal It's time to close this issue, don't you think? ;)

Wait, please.
I'll close it when I am done with the handler. I'm afraid I can't do it without help.

**Comment by cleargoal** [comment]
> > Is `/usr/bin` the directory you mean?
> 
> You missed the `~` (tilde) sign. Path `~/` means user home directory, which in your case will be automagically translated to `/home/yefrem/`
> 
> Give it a try:
> 
> ```shell
> realpath ~/
> ```

![image](https://github.com/user-attachments/assets/58e89a1a-4efc-47c9-8097-c1200d5c6dc1)
This is my user home directory.
Did I understand you and correctly create a 'bin' directory in the proper place?

**Comment by rozwell** [comment]
> This is my user home directory. Did I understand you and correctly create a 'bin' directory in the proper place?

Yes, you can confirm what's inside that directory with the command: 
```bash
ls -l ~/bin/
```

**Comment by cleargoal** [comment]
> Yes, you can confirm what's inside that directory with the command:
> 
> ```shell
> ls -l ~/bin/
> ```

Thank you!
Did you run these commands?
```
desktop-file-install phpstorm-url-handler.desktop
update-desktop-database
```
Would you happen to have that .desktop label anywhere?

I ran commands but didn't see any visuals.
